### ISSUE TITLE
CI failing after recent merges

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-spec#1034]
Runs of CI for the last 2 weeks on the development/v3.0.1 branch fail with the following error:

```
Run mike set-default v3.0
  mike set-default v[3](https://github.com/spdx/spdx-spec/actions/runs/10325067391/job/28595098194#step:14:3).0
  shell: sh -e {0}
Config value 'nav': Expected nav to be a list, got None
error: Aborted with a configuration error!
Error: Process completed with exit code 1.
```

It looks like all the CI runs for this workflow `.github/workflows/publish_v3.yml` have failed for the development/v3.0.1 branch.

Note that the pull request CI's are passing, it is just after merge we're getting the failures.

**Comment by goneall** [comment]
@bact @zvr - any ideas?  I looked at the mkdocs config files and didn't see anything obvious.

**Comment by bact** [comment]
It's because `mike` is called without passing the customized mkdocs.yml to it.

It used to be the case that we call the standard mkdocs.yml.

But since now we're using a dynamic list of model files, the list have to be added to mkdocs.yml before it can be used.
Lines 56-65, do exactly that, they combine docs/model/mkdocs-files.yml and mkdocs.yml, to have mkdocs-final.yml:

https://github.com/spdx/spdx-spec/blob/001f4673ff960fb58d37649c08a444160556bebb/.github/workflows/publish_v3.yml#L56-L65

Then `mike deploy` in Line 69, call mike with this customized config, passing option `--config-file "$FINAL_MKDOCS_YML"`:

https://github.com/spdx/spdx-spec/blob/001f4673ff960fb58d37649c08a444160556bebb/.github/workflows/publish_v3.yml#L69

What happened is, I forgot to pass that option to `mike set-default`, at Line 74:

https://github.com/spdx/spdx-spec/blob/001f4673ff960fb58d37649c08a444160556bebb/.github/workflows/publish_v3.yml#L74

Thus the error.

The fix is purposed in PR #1015, together with a fix in canonical URL setting.

**Comment by goneall** [comment]
Fixed
